### ISSUE TITLE
BUG: Geometry column assigned after creation of gdf is not GeometryArray

### ISSUE THREAD

**Issue by martinfleis** [geopandas/geopandas#1249]
Consider following scenario (taken from OSMnx code) of gdf creation.

```
x = [0, 1]
y = [2, 3]
gdf = gpd.GeoDataFrame([x, y], columns=['x', 'y'])
gdf['geometry'] = gdf.apply(lambda row: Point(row['x'], row['y']), axis=1)
```

In this case, resulting gdf.geometry is not GeometryArray. Which means that after #1177 `unary_union` does not work among other things, but geopandas pretends that proper geometry is there and most of the things work as expected.

```
gdf.geometry

0    POINT (0 1)
1    POINT (2 3)
Name: geometry, dtype: object
```

You have to use `set_geometry` to convert it to GeometryArray.

```
gdf2 = gdf.set_geometry('geometry')
gdf2.geometry

0    POINT (0.00000 1.00000)
1    POINT (2.00000 3.00000)
Name: geometry, dtype: geometry
```

I know that it is not ideal way of GeoDataFrame creation (gdf is invalid at first), but it is apparently used. I think we should check if the new column called 'geometry' is created and use `set_geometry` method to make sure everything works as intended. #1177 is causing OSMnx to fail with geopandas master (cc @gboeing).

**Comment by gboeing** [comment]
Interesting. I'll try to keep my eye on this thread. Let me know if this geopandas feature gets released *without* the check proposed by @martinfleis and I'll try to patch OSMnx quickly if needed.

**Comment by martinfleis** [comment]
Just realised that this is a duplicate of #1194, so closing here.
